astropy/cosmology/core.py

```diff
@@ -125,9 +143,18 @@ def __init_subclass__(cls):
         cls.__all_parameters__ = cls.__parameters__ + tuple(derived_parameters)
 
         # -------------------
-        # register as a Cosmology subclass
+        # Register as a Cosmology subclass
+
+        if skip_registration:
+            return
+
         _COSMOLOGY_CLASSES[cls.__qualname__] = cls
 
+        # register YAML
```

Comment:
Not a real issue, but I think I would move all the registration to a new class method, say `_register_class`. Indeed, with that a simpler solution to the problem here is just to override
```
@classmethod
def _register_class(cls):
    pass
```

Reply:
I like this suggestion! Implementing now.